### PR TITLE
[DDCE-4454] Upgraded hmrc-mongo + dependencies

### PR DESCRIPTION
**Pull request by matthewwardcap**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,3 +1,19 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
```
Comment (frikit) [inline build.sbt:2]: delete license header in .sbt files as it is not required

File: project/plugins.sbt [hunk 2]
```diff
@@ -1,5 +1,5 @@
-resolvers += "HMRC-open-artefacts-maven" at "https://open.artefacts.tax.service.gov.uk/maven2"
-resolvers += Resolver.url("HMRC-open-artefacts-ivy", url("https://open.artefacts.tax.service.gov.uk/ivy2"))(Resolver.ivyStylePatterns)
+resolvers += "HMRC-open-artefacts-maven2" at "https://open.artefacts.tax.service.gov.uk/maven2"
+resolvers += Resolver.url("HMRC-open-artefacts-ivy2", url("https://open.artefacts.tax.service.gov.uk/ivy2"))(Resolver.ivyStylePatterns)
 resolvers += Resolver.typesafeRepo("releases")
```
Comment (frikit) [inline project/plugins.sbt:3]: can be deleted

**Review by frikit — CHANGES_REQUESTED** [review]
- Fix in build.sbt `test:scalastyle` to `Test/scalastyle`
- Fix in build.sbt `it:scalastyle` to `IntegrationTest/scalastyle`
- Fix in run_all_tests.sh `it:test`

File: build.sbt (hunk 1 above)
Comment (matthewwardcap) [reply build.sbt:2]: Done

File: project/plugins.sbt (hunk 2 above)
Comment (matthewwardcap) [reply project/plugins.sbt:3]: Done

File: project/AppDependencies.scala [hunk 3]
```diff
@@ -1,32 +1,47 @@
...
     "uk.gov.hmrc"       %% "bootstrap-frontend-play-28"     % bootstrapVersion,
     "com.typesafe.play" %% "play-json-joda"                 % "2.9.4",
-    "uk.gov.hmrc"       %% "domain"                         % "8.1.0-play-28"
+    "uk.gov.hmrc"       %% "domain"                         % "8.3.0-play-28"
   )
 
   val test: Seq[ModuleID] = Seq(
     "uk.gov.hmrc"             %% "bootstrap-test-play-28"   % bootstrapVersion,
-    "com.typesafe.play"       %% "play-test"                % PlayVersion.current,
     "org.scalatestplus.play"  %% "scalatestplus-play"       % "5.1.0",
```
[the remark this answers is not in the document]
Comment (matthewwardcap) [reply project/AppDependencies.scala]: Done

**Pull request merged**